### ISSUE TITLE
Revisit Boost.Stacktrace backend implementation

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#6146]
The basic stacktrace implementation used by Boost.Stacktrace does not provide information such as file paths or line numbers.

We need to determine which backend implementation to use. The options are:
- libbacktrace
- addr2line

libbacktrace provides the most useful information for debugging however, it is related to GCC and does not appear to be located in a standard location. This makes relying on this library difficult because Boost.Stacktrace needs to be told where the `<backtrace.h>` header is.

addr2line provides good information, but doesn't always provide file paths and line numbers. This library is easy to get working if the necessary packages are installed.

See the Boost documentation for more information. As of this writing, the current version of Boost is 1.78.

**Comment by trel** [comment]
> To explicitly specify a path to the <backtrace.h> header you could define the BOOST_STACKTRACE_BACKTRACE_INCLUDE_FILE to a full path to the header.

You're saying the fullpath to `backtrace.h` behaved more poorly than using a relative path and adding `/usr/lib/gcc/x86_64-linux-gnu` to the include directories?

**Comment by korydraughn** [comment]
Yes. The path was incorrect after being injected into the source.

**Comment by trel** [comment]
Boost proper might want to hear about that themselves.

**Comment by korydraughn** [comment]
That's true. I'll write up an issue for them if I don't come across an answer soon.

**Comment by korydraughn** [comment]
Here is the libbacktrace repo: https://github.com/ianlancetaylor/libbacktrace

We could make it an external if we wanted to. Just something to think about.

**Comment by trel** [comment]
Indeed.  I defer.   Perhaps with a little more usage ourselves, we'll have a stronger opinion.

**Comment by SwooshyCueb** [comment]
This may be made easier by the switch to libstdc++. Will investigate.

**Comment by SwooshyCueb** [comment]
So it looks like EL doesn't include libbacktrace as part of its gcc packages. Or any of its packages for that matter. If we want to use the libbacktrace backend, we will have to add libbacktrace as an externals package. I may end up doing this as part of my work on Ubuntu 24.04, as some of the stack traces I've been seeing in the logs recently have been less than helpful.
